### ISSUE TITLE
DFS support for VHT160 missing from the 5.15 based version?

### ISSUE THREAD

**Issue by hnyman** [greearb/ath10k-ct#192]
I think that the DFS support for VHT160 is missing from the 5.15 based ath10k-ct.

DFS worked ok with my previous OpenWrt master build with 5.10 ath10k-ct two days ago, but now with 5.15 ath10-ct there are again similar errors as there were earlier (before @greearb patched DFS support in ath10k-ct):

```
Sat Nov 27 22:23:30 2021 daemon.notice hostapd: wlan0: interface state HT_SCAN->DFS
Sat Nov 27 22:23:30 2021 daemon.notice hostapd: wlan0: DFS-CAC-START freq=5500 chan=100 sec_chan=1, width=2, seg0=114, seg1=0, cac_time=60s
Sat Nov 27 22:23:30 2021 daemon.err hostapd: DFS start_dfs_cac() failed, -1
Sat Nov 27 22:23:30 2021 daemon.err hostapd: Interface initialization failed
Sat Nov 27 22:23:30 2021 daemon.notice hostapd: wlan0: interface state DFS->DISABLED
Sat Nov 27 22:23:30 2021 daemon.notice hostapd: wlan0: AP-DISABLED
```

The scan fails at the start.

Doing the DFS scan with VHT80 works ok.

Looking at the patch https://github.com/greearb/ath10k-ct/commit/23c74615877fa6bf2c09df773a42e30073657cb8
it looks to me  that `ieee80211_iface_combination ath10k_10_4_ct_if_comb` in mac.c was patched for the 5.10 and 5.11 versions
but that the new mac.c in the new 5.15 version is still missing `80P80` and `160` lines:
https://github.com/greearb/ath10k-ct/commit/23c74615877fa6bf2c09df773a42e30073657cb8#diff-5f6798ba0e5c63770dac9457de5066fdd5d0d71f96aae57ca35b0b293776e279R10895-R10900

Clear difference to what was done for 5.10:
https://github.com/greearb/ath10k-ct/commit/23c74615877fa6bf2c09df773a42e30073657cb8#diff-f24773d2337658245c7a0b65aadb66c15c756886dadf9b8b05f4ea5f6338be76R10779-L10784

Might be that I do not understand the supposed contents right, but that looks like a possible bug in ath10k-ct to me. Possibly the mac.c was copied to 5.15 (and 5.13?) before it was fixed in 5.10 and 5.11.

This patch (for OpenWrt) seems to fix things:

package/kernel/ath10k-ct/patches/010-fix-DFS-VHT160.patch

```
--- a/ath10k-5.15/mac.c
+++ b/ath10k-5.15/mac.c
@@ -10896,7 +10896,9 @@
 		.radar_detect_widths =	BIT(NL80211_CHAN_WIDTH_20_NOHT) |
 					BIT(NL80211_CHAN_WIDTH_20) |
 					BIT(NL80211_CHAN_WIDTH_40) |
-					BIT(NL80211_CHAN_WIDTH_80),
+					BIT(NL80211_CHAN_WIDTH_80) |
+					BIT(NL80211_CHAN_WIDTH_80P80) |
+					BIT(NL80211_CHAN_WIDTH_160),
 #endif
 	},
 };
```

Log extract:
```
Sat Nov 27 23:37:29 2021 daemon.notice hostapd: wlan0: interface state HT_SCAN->DFS
Sat Nov 27 23:37:29 2021 daemon.notice hostapd: wlan0: DFS-CAC-START freq=5500 chan=100 sec_chan=1, width=2, seg0=114, seg1=0, cac_time=60s
Sat Nov 27 23:37:29 2021 daemon.notice netifd: Wireless device 'radio0' is now up
Sat Nov 27 23:38:31 2021 daemon.notice hostapd: wlan0: DFS-CAC-COMPLETED success=1 freq=5500 ht_enabled=0 chan_offset=0 chan_width=5 cf1=5570 cf2=0
```

**Comment by hnyman** [comment]
Fixed with dc350bbf41

**Comment by hnyman** [comment]
The downstream fix at OpenWrt has been merged. Thanks for the fix here.

For the sake of completeness, please consider also fixing your other "new" versions:
the 5.12 and 5.13 versions are similarly faulty as 5.15 was. 
80P80 and 160 are missing from the same place in their mac.c

(I am not sure what their actual role is / will be, but having deviation there may lead into some sneaky problems later)
